src/main/java/com/google/step/youtube/DescriptionParser.java

```diff
@@ -52,14 +52,34 @@ public Pattern getPattern() {
      * @param description of the YouTube video to be parsed.
      * @return A list of all matches found in description.
      */
-    public static List<String> findMatches(Pattern pattern, String description) {
-        List<String> matches = new ArrayList<>();
+    public static List<OfferSnippet> findMatches(Pattern pattern, String description) {
+        List<OfferSnippet> matches = new ArrayList<>();
         Matcher matcher = pattern.matcher(description);
 
         while (matcher.find()) {
-            matches.add(matcher.group());
+            matches.add(OfferSnippet.create(matcher.group(), getSnippet(matcher.start(), description)));
         }
         return matches;
     }
 
+    /**
+     * Finds the snippet (paragraph) of description which conatins the promocode index.
+     * 
+     * @param promocodeIndex The index in the description which marks the start of the promocode.
+     * @param description of the video from which the promocode was extracted.
+     * @return The snippet (paragraph) which contains the promocodeIndex. A paragraph delineated by 
```

Comment:
Could you add a maximum value.  What if the paragraph never ends?

Reply:
Added. Also changed to line instead of paragraph to account for lists of links in descriptions.